feature_engine/base_transformers.py

```diff
@@ -144,3 +144,26 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
     # for the check_estimator tests
     def _more_tags(self):
         return _return_tags()
+
+    def get_feature_names(
+        self,
+        input_features: List[str] = None,
+    ) -> List:
+        """
+        Return feature names for output features.
+
+        Parameters
+        ----------
+        input_features : list of str of shape (n_features,)
+            String names for input features if available. By default,
+            self.variables_ is used
```

Comment:
self.variables_ contains the variables that are modified by the transformer. Not all the variables in the dataframe. Why would this be the default?

Reply:
Yes, this is a problem. I didn't want to modify the internal behaviour of the encoders before I showed it to you. 

I could use a different variable to keep every column in the input on `fit`, and use that as the basis to determine the output features.

Reply:
this particular transformer captures only numerical variables by default or checks that the variables entered by the user are of type numerical.

I think the methods wants to capture all variables in the dataset, so list(X.columns) should do. But as per my previous comment, I think an attribute is enough, we don't need a  method. And also, would like to understand more about the value of this attribute. Because we basically need to modify all transformers. It is some work

Reply:
d452ef2

Reply:
I think these are looking quite simple now, it would be nice if we could add them to all transformer at once?

Reply:
> 
> 
> I think these are looking quite simple now, it would be nice if we could add them to all transformer at once?

That would be great!